### ISSUE TITLE
demo throws MediaError {code: 4, message: 'MEDIA_ELEMENT_ERROR: Empty src attribute'}

### ISSUE THREAD

**Issue by guest271314** [davedoesdev/webm-muxer.js#4]
demo throws 

`MediaError {code: 4, message: 'MEDIA_ELEMENT_ERROR: Empty src attribute'}`

**Comment by davedoesdev** [comment]
Thanks for the bug report, appreciate it. I'll see if I can repro it when I get to look at it, hopefully later this week.
Which browser/version are you using please?

**Comment by guest271314** [comment]
Chromium 94.0.4587.0

**Comment by davedoesdev** [comment]
Thanks, waiting for Linux dev channel to update so I can test

**Comment by guest271314** [comment]
> Thanks, waiting for Linux dev channel to update so I can test

https://download-chromium.appspot.com/

**Comment by davedoesdev** [comment]
Thanks for the link. With time constraints I'm happy to test against dev but not bleeding edge.

**Comment by guest271314** [comment]
"Bleeding edge" is where the rub is. WebCodecs specification is still `'FLTP'` for `'format'` of `AudioData`. `'FLTP'` was only implemented for at most several months before Chromium started supporting `'f32-planar'` which I located reading Chromium commits. WebCodecs is a moving target right now in both specification and implementation. I am willing to test the code that you have for a fix. 

I have been conceiving of ways to support audio for webm-writer-js (https://github.com/thenickdude/webm-writer-js/issues/1, https://github.com/kbumsik/opus-media-recorder/issues/23). Your project, specifically the commands you are using  might work for that purpose.

Instead of using WASM which requires substantial RAM and not infrequently crashes the tab, ideally I will use the commands you use to libwebm and webmtools  in WASM directly from the browser to the applications you are using to encode using, e.g., both Native Messaging https://github.com/guest271314/native-messaging-espeak-ng and WebTransport .https://github.com/guest271314/webtransport. 

I previously used mkvmerge https://github.com/guest271314/native-messaging-mkvmerge to merge multiple audio and video tracks into a single WebM container with the deprecated Chrome Apps, I intend to update that code to use MV3 extension.

I am interested in the actual code you are running for WASM.

**Comment by davedoesdev** [comment]
OK testing the demo with 94.0.4597.0 I got:

> Uncaught (in promise) DOMException: Could not start audio source

I'll take a look when I get some spare time, hopefully this week.

**Comment by davedoesdev** [comment]
Looks like a general audio problem with that build (other sites using getusermedia fail with same error)

**Comment by davedoesdev** [comment]
e.g. https://www.onlinemictest.com/ works in Chrome 93 Linux but not Chrome 94

**Comment by davedoesdev** [comment]
OK I can run with `/chrome --disable-features=AudioServiceOutOfProcess` (from https://bbs.archlinux.org/viewtopic.php?id=268160 - seems to the usual pulse/also linux woe).

Then I can repro your error when I click stop:

demo.js:2 MediaError {code: 4, message: 'MEDIA_ELEMENT_ERROR: Empty src attribute'}

**Comment by guest271314** [comment]
Got the demo working. 

Occasionally the recorded audio output is faster than expected. The audio track is not listed by `mkvmerge -J`. `mediainfo` lists 

```
Channel(s)                               : 1 channel
Sampling rate                            : 48.0 kHz
```
which could be causing the faster than expected playback rate. I used 6000 bitrate and 22050 sample rate for `AudioEncoder` for 1 channel https://plnkr.co/edit/nb1O08jar2J5Y6fQ?preview. 

See also issues with `MediaStreamTrackGenerator` "overflow" https://bugs.chromium.org/p/chromium/issues/detail?id=1184070, https://bugs.chromium.org/p/chromium/issues/detail?id=1199377 which can also produce a faster encoding and resulting playback rate than expected. I will continue testing.

**Comment by davedoesdev** [comment]
@guest271314 how did you fix the MediaError?

**Comment by davedoesdev** [comment]
IIRC the error from setting src to empty string is unavoidable

**Comment by davedoesdev** [comment]
e.g. https://github.com/google/shaka-player/issues/80 doesn't suggest an alternative

**Comment by guest271314** [comment]
I did not fix the media error. Given there are no instructions I opened the console and noticed the error, continued trying the site. Eventually a video appears, or does not appear, I checked "Record" and a video downloaded. Then I tested at least 15 more times since. The result of the audio is not consistent. I suggest testing audio several times at least a few dozen to determine if you can reproduce the faster than expected recording and playback rate.

**Comment by davedoesdev** [comment]
This may help, I'll try it: https://github.com/google/shaka-player/commit/0e3d10e61cb08cc35750e48face4e58f24542fc8

**Comment by davedoesdev** [comment]
I'll try the audio rate to see what I get

**Comment by davedoesdev** [comment]
I've opened #5 to add demo instructions and #6 to investigate audio rate, thanks!

**Comment by guest271314** [comment]
The error is not an issue as long as the code produces consistent result. Instructions and what to expect on the demo page would be helpful.

I tested with my own voice and 

```
$ google-chrome-unstable \
--enable-experimental-web-platform-features \
--allow-file-access-from-files \
 --autoplay-policy=no-user-gesture-required \
--use-fake-device-for-media-stream \
--use-fake-ui-for-media-stream \
--use-file-for-fake-audio-capture=input.wav \
--user-data-dir=/tmp \
https://rawgit-now.netlify.app/davedoesdev/webm-muxer.js/main/demo.html
```

where input.wav is file from 

`espeak-ng -w input.wav 'Test'`.

**Comment by davedoesdev** [comment]
This issue fixed by 4b9a10903fb36e36db97d7572a501c2d1c80c03f
